src/_common/scroll/inview/inview.ts

```diff
@@ -1,76 +1,62 @@
 import Vue, { CreateElement } from 'vue';
-import { Component, Prop, Watch } from 'vue-property-decorator';
-import { arrayRemove } from '../../../utils/array';
-import { findVueParent } from '../../../utils/vue';
-import { Scroll } from '../scroll.service';
-import { ScrollInviewContainer } from './container';
+import { Component, Emit, Prop, Watch } from 'vue-property-decorator';
+import { findRequiredVueParent } from '../../../utils/vue';
 import { AppScrollInviewParent } from './parent';
 
-// This is the root container we use if there's no inview parent.
-let BaseContainer: ScrollInviewContainer;
-if (!GJ_IS_SSR) {
-	BaseContainer = new ScrollInviewContainer(Scroll.watcher);
-}
-
 @Component({})
 export class AppScrollInview extends Vue {
 	@Prop({ type: String, default: 'div' })
 	tag!: string;
 
-	@Prop({ type: Number, default: 0 })
-	extraPadding!: number;
-
+	/**
+	 * The margin will effectively get added to the bounding box of this element
+	 * to make it larger or smaller. It should be in the format of a CSS margin
+	 * property and always have the "px" after each value.
```

Comment:
Just so you know, this field can look like any one of these:
- `10px` to make all sides 10px
- `10px 0px` to make just the top/bottom 10px
- `10px 0px 5px 0px` to make the top 10px, right 0, bottom 5px, left 0.

It follows the API that IntersectionObserver takes, which mimics the CSS margin property rules. It can only be px or % based and from my testing each value needs a unit.

I didn't want to add extra complexity by trying to validate this string ourselves since the browser throws a DOM error when it's an invalid input.